src/arti/tor-dirmgr/src/authority.rs

```diff
@@ -44,6 +46,29 @@ impl Authority {
     }
 }
 
+impl FromStr for Authority {
+    type Err = Error;
+
+    /// Parse Authority from a string.
+    fn from_str(authority_raw: &str) -> Result<Self, Self::Err> {
+        // name, v3ident_raw
+        let authority: Vec<&str> = authority_raw.split_whitespace().collect();
+
+        if authority.len() != 2 {
+            bail!("Invalid format for authority.");
+        }
```

Comment:
```suggestion
        assert_eq!(authority.len(), 2,"Invalid format for authority.");
```
Also needs the import, if we want this.

@tharvik or is a panic too much here? Is there a way to have an assert with an error? Something like

```
authority.len() == 2 || bail!("");
```

?

Reply:
Are we bailing out of `anyhow`?

Reply:
@ineiti never panic (cf H2G2)

We can use `snafu`, a personal favorite, which has `ensure` (a way to assert and return early), but we have `anyhow`, let's stick with it.

Reply:
OK, let's keep the `if` then...